gensim/matutils.py

```diff
@@ -382,6 +383,93 @@ def cossim(vec1, vec2):
     return result
 
 
+def kullback_leibler(vec1,vec2,num_of_features=None):
+    """
+    A similarity metric between two probability distributions.
+    Returns a similarity in range <0,1> where values closer to 0 mean a higher similarity.
+    Uses the scipy.stats.entropy method to identify kullback_leibler convergence value.
+    If the distribution draws from a certain number of docs, that value must be passed.
+    """
+    if scipy.sparse.issparse(vec1) and scipy.sparse.issparse(vec2):
```

Comment:
Yes, if the performance is good (=not much slowdown, let's say not more than 20%), we can do that. Another option: more code paths for the various input type combinations.